### PR TITLE
fix for conversion micro unit in modification by filter

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

**Review by etiennehomer — CHANGES_REQUESTED** [review]
In the equipments spreadshit, the susceptance is in μS. So it's normal to use unitToMicroUnit() to convert them. The problem is that for modification by formula and modication by filter, the susceptance value is taken as Siemens, not μSiemens

File: src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts [hunk 1]
```diff
@@ -150,15 +163,32 @@ export const FIELD_OPTIONS: {
     R: {
         id: FieldType.R,
         label: 'SeriesResistanceText',
+        unit: MICRO_SIEMENS,
```
Comment (etiennehomer) [inline src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts:166]: Ohm for resistance ! Do not change here

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts (hunk 1 above)
Comment (ghazwarhili) [reply src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts]: good catch !

File: src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts [hunk 2]
```diff
@@ -130,12 +137,18 @@ export const FIELD_OPTIONS: {
...
         dataType: DataType.DOUBLE,
+        outputConverter: (value) => unitToKiloUnit(value),
+        inputConverter: (value) => kiloUnitToUnit(value),
     },
     HIGH_SHORT_CIRCUIT_CURRENT_LIMIT: {
         id: FieldType.HIGH_SHORT_CIRCUIT_CURRENT_LIMIT,
         label: 'HighShortCircuitCurrentLimit',
+        unit: KILO_AMPERE,
         dataType: DataType.DOUBLE,
+        outputConverter: (value) => unitToKiloUnit(value),
```
Comment (etiennehomer) [inline src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts:150]: Conversion doesn't work for short circuit current limit for me

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts [hunk 3]
```diff
@@ -130,12 +137,18 @@ export const FIELD_OPTIONS: {
     LOW_SHORT_CIRCUIT_CURRENT_LIMIT: {
         id: FieldType.LOW_SHORT_CIRCUIT_CURRENT_LIMIT,
         label: 'LowShortCircuitCurrentLimit',
+        unit: KILO_AMPERE,
         dataType: DataType.DOUBLE,
+        outputConverter: (value) => unitToKiloUnit(value),
```
Comment (etiennehomer) [inline src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-constants.ts:142]: To inverse

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-utils.ts [hunk 4]
```diff
@@ -19,7 +19,23 @@ import { Assignment, DataType } from './assignment.type';
 import { FIELD_OPTIONS } from './assignment-constants';
 
 export const getDataType = (fieldName?: string | null) => {
-    return Object.values(FIELD_OPTIONS).find((fieldOption) => fieldOption.id === fieldName)?.dataType;
+    return getFieldOptionType(fieldName)?.dataType;
+};
+
+export const getFieldOptionType = (fieldName?: string | null) => {
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-utils.ts]: done

File: src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-utils.ts [hunk 5]
```diff
@@ -19,7 +19,23 @@ import { Assignment, DataType } from './assignment.type';
...
+    return Object.values(FIELD_OPTIONS).find((fieldOption) => fieldOption.id === fieldName);
+};
+
+type FieldValue = string | number | boolean;
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-utils.ts]: done

File: src/components/dialogs/network-modifications/by-filter/by-assignment/modification-by-assignment-dialog.tsx [hunk 6]
```diff
@@ -83,6 +93,8 @@ const ModificationByAssignmentDialog: FC<any> = ({
         (formData: ModificationByAssignment) => {
             const assignmentsList = formData[ASSIGNMENTS].map((assignment) => {
                 const dataType = getDataType(assignment[EDITED_FIELD]);
+                const valueConverted = convertOutputValue(assignment[EDITED_FIELD], assignment[VALUE_FIELD]);
+                assignment[VALUE_FIELD] = Number(valueConverted);
```
Comment (etiennehomer) [inline src/components/dialogs/network-modifications/by-filter/by-assignment/modification-by-assignment-dialog.tsx:97]: Add the VALUE_FIELD in the return to be homogeneous with the useEffet() above.

Comment (etiennehomer) [reply src/components/dialogs/network-modifications/by-filter/by-assignment/modification-by-assignment-dialog.tsx:97]: And remove Number()

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**